internal/comparetest/compare.go

```diff
@@ -267,3 +268,161 @@ func numberTypeToString(t pmetric.NumberDataPointValueType) string {
 		return "none"
 	}
 }
+
+// CompareLogs compares each part of two given Logs and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareLogs(expected, actual plog.Logs) error {
+	if expected.ResourceLogs().Len() != actual.ResourceLogs().Len() {
+		return fmt.Errorf("amount of ResourceLogs between Logs are not equal (expected: %d, actual: %d)",
+			expected.ResourceLogs().Len(),
+			actual.ResourceLogs().Len())
+	}
+
```

Comment:
We should make copies of plog.Logs and sort all the slices as it's done for metrics. Right?

Reply:
@jackycsl, do you plan on iterating on this PR?